src/components/ActionModal/ActionModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Modal, ModalProps, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+import { css } from '@patternfly/react-styles';
+import { PropsWithChildren, ReactElement } from 'react';
+
+import { useRefresh } from 'hooks/useRefresh';
+import { DataValues, IServiceContainer } from 'hooks/useServiceContainer';
+
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+
+import styles from './ActionModal.module.css';
+
+interface IActionModalProps {
+  modalTitle: string;
+  actionTitle: string;
+  isOpen: boolean;
+  isSubmitDisabled: boolean;
+  onToggle: () => void;
+  onSubmit: () => void;
+  serviceContainerCreatingUpdating: IServiceContainer;
+  modalVariant: ModalProps['variant'];
+  onSuccessActions?: ReactElement[];
+}
+
+/**
+ * Modal used for the creating or updating action.
+ *
+ * @param modalTitle - title of the modal
+ * @param actionTitle - title of the confirm action button
+ * @param isOpen - is the modal open?
+ * @param isSubmitDisabled - is the confirm action button disabled?
+ * @param onToggle - function toggling the modal visibility
+ * @param onSubmit - confirm action button callback
+ * @param serviceContainerCreatingUpdating - service container of the confirm action
+ * @param modalVariant - modal size variant
+ * @param onSuccessActions - additional action buttons to be displayed after susccessful confirm action
+ */
+export const ActionModal = ({
+  children,
+  modalTitle,
+  actionTitle,
+  isOpen,
+  isSubmitDisabled,
+  onToggle,
+  onSubmit,
+  serviceContainerCreatingUpdating,
+  modalVariant,
+  onSuccessActions,
+}: PropsWithChildren<IActionModalProps>) => {
+  const refresh = useRefresh();
+
+  const wasActionSuccessful =
+    serviceContainerCreatingUpdating.data !== DataValues.notYetData && !serviceContainerCreatingUpdating.error;
+
+  const onClose = () => {
+    if (wasActionSuccessful) {
+      refresh();
+    } else {
+      onToggle();
+    }
+  };
+
+  return (
+    <Modal
+      variant={modalVariant}
+      isOpen={isOpen}
+      onClose={onClose}
+      actions={[
+        <Button variant="primary" onClick={onSubmit} isDisabled={isSubmitDisabled || serviceContainerCreatingUpdating.loading}>
+          {!serviceContainerCreatingUpdating.error && (
+            <ServiceContainerLoading variant="icon" {...serviceContainerCreatingUpdating} title={actionTitle} />
+          )}{' '}
+          {actionTitle}
+        </Button>,
+        ...(wasActionSuccessful && onSuccessActions ? onSuccessActions : []),
+        <Button variant="link" onClick={onClose}>
+          {wasActionSuccessful ? 'Close and refresh' : 'Cancel'}
```

Comment:
There is one more scenario, when this is not working properly:
- first request is successful (ie. NEW -> DELETED)
- second request fails (ie. DELETED -> NEW  // invalid operation)

Then user can see only "Cancel" and Close icon, in both cases, the user will be returned to the deprecated page.

Solution would be to keep one more useState variable that would be marked as `true` when `wasActionSuccessful` is true for the first time.

So basically there will be two states captured:

- `wasLastActionSuccessful` (use instead of `wasActionSuccessful`)
- `wasAnyActionSuccessful`